pages/client-libraries/rust.mdx

```diff
@@ -0,0 +1,714 @@
+let _read_path = "MATCH p=(d:Developer)-[r:LOVES]->(t:Technology) RETURN p";
+let _columns = connection.execute(_read_path, None);
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Path(path) =>
+                {
+                    println!("Path nodes: {:?}", path.nodes);
+                    println!("Path relationships: {:?}", path.relationships);
+                }
+                _ => continue
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+Path will contain [Nodes](#process-the-node-result) and [Relationships[#process-the-relationship-result], that can be accessed in the same way as in the previous examples, by casting them to the relevant type.
+
+#### Types mapping and casting
+
+ Here is the full table of the mapping between Memgraph Cypher types and the types used in the Rust driver:
+
+|  Cypher Type | Driver Type        |
+|-------------:|:-------------------|
+|          Null| Null               |
+|       String | String             |
+|      Boolean | bool               |
+|      Integer | i64                |
+|        Float | f64                |
+|         List | Vec< Value >       |
+|          Map | HashMap< String, Value >|
+|         Node | Node               |
+| Relationship | Relationship       |
+|         Path | Path               |
+| UnboundRelationship | UnboundRelationship |
+|     Duration | Duration           |
+|         Date | NaiveDate          |
+|    LocalTime | NaiveTime          |
+|LocalDateTime | NaiveDateTime      |
+
+Keep in mind that Memgraph does not support timezones at the moment. 
+
+### Transaction management
+
+Transaction is a unit of work that is executed on the database, it could be some basic read, write or complex set of steps in form of series of queries. There can be multiple ways to mange transaction, but usually, they are managed automatically by the driver or manually by the explicit code steps. 
+Transaction management defines how to handle the transaction, when to commit, rollback, or terminate it.
+
+Currenty, there are two ways to manage transactions in the Rust driver:
+
+- [Manual transaction management](#manual-transaction-management)
+- [Implicit transaction management](#implicit-transaction-management)
+
+
+#### Manual transaction management
+
+Once the connection is established, you can run queries via the following methods:
+
+- `execute` - Starts the transaction for executing a query, returns the result.
+- `execute_without_results` - Executes the query without returning the result.
+
+Only the execute will be manually managed by the user, and the user will need to commit or rollback the transaction. This means the transacition is started inside the `execute` method via `BEGIN`, and it will be finished with the `COMMIT` or `ROLLBACK`, depending on the logic. 
+Here is the example:
```

Comment:
```suggestion
Only the `execute()` method should be manually managed by the user, meaning you need to commit or roll back the transaction. This means the transaction is started inside the `execute` method when it `BEGIN`, and it will be finished with the `COMMIT` or `ROLLBACK`, depending on the logic. 
Here is the example:
```